lib/xclarity_client/xclarity_base.rb

```diff
@@ -14,17 +14,18 @@ def initialize(conf, uri)
     end
 
     def connection_builder(conf, uri)
-      $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Creating connection to #{conf.host + uri}" 
-      # Building configuration, convert to https if protocol not specified
+      $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Building the url"
+      #Building configuration
       hostname = URI.parse(conf.host)
-      unless hostname.scheme
-        hostname = URI::HTTPS.build({ :host     => hostname.host,
-                                      :port     => hostname.port,
-                                      :path     => hostname.path,
-                                      :query    => hostname.query,
-                                      :fragment => hostname.fragment })
-      end
-      @conn = Faraday.new(:url => "#{hostname}#{uri}") do |faraday|
+      url = URI::HTTPS.build({ :host     => hostname.scheme ? hostname.host : hostname.path,
+                               :port     => conf.port,
```

Comment:
Validate this port when mount the URL, because it can raise stranges errors in logs.